### DISCUSSION TITLE
Mailer not working for data-importer?

### DISCUSSION THREAD

**Discussion by psychowood** [firefly-iii/firefly-iii#8309] [General]
Hi all,
I'm opening a discussion here since I'm not really sure that this is not a problem on my side rather than a bug, but I've been struggling with this for a few hours and I can't seem to find a solution, so here I am.

Context:
- firefly-iii and data-importer are configured via docker-compose, working as expected besides this issue
- both apps are in the same compose file, in the same network, behind the same reverse proxy (I also tried by exposing data-importer directly from the docker host, same result)
- firefly-iii mail are working correctly, I can send and receive the test email
- data-importer has exactly the same configuration of firefly-iii (w/ different names for env variables where needed of course) in regard of mail host, user, password, from account and destination account (I set the same email of the user in firefly-iii).

Thing is, I can't seem to trigger the report. Even stranger, there's no trace of anything mail related in the logs (I enabled all the debug modes I could find), not even the message ['Now in sendReportOverMail'](https://github.com/firefly-iii/data-importer/blob/6852b1af084d39836121688d92f62d95e355d8f8/app/Handlers/Events/ImportedTransactionsEventHandler.php#L40) I'd expect from the source code.

exim4 is not reporting any connection from data-importer, while it does for firefly-iii.

I tried both the 1.4.0 version and the 1.3.12 of data-importer, same result. Changing the mailer to 'log' does not change the behaviour.

Am I missing something?

Thanks, and happy new year :)

----

For reference, these are the relevant variables reported from the running data-importer container by portainer:

```
ENABLE_MAIL_REPORT 	true
LOG_LEVEL 	debug
LOG_RETURN_JSON 	true
MAIL_DESTINATION 	xxxxx@mydomain
MAIL_ENCRYPTION 	null
MAIL_FROM_ADDRESS 	yyyyy@mydomain
MAIL_HOST 	exim4
MAIL_MAILER 	smtp
MAIL_PASSWORD 	null
MAIL_PORT 	8025
MAIL_USERNAME 	null
```

while these are the ones for firefly-iii:

```
MAIL_ENCRYPTION 	null
MAIL_FROM 	yyyy@mydomain
MAIL_HOST 	exim4
MAIL_MAILER 	smtp
MAIL_PASSWORD 	null
MAIL_PORT 	8025
MAIL_SENDMAIL_COMMAND 	
MAIL_USERNAME 	null
```

**Comment by JC5** [comment]
How do you run the import routine? Over the UI, or using the command line?

**Reply by psychowood** [reply]
UI, importing from CSV. Tested different file formats,  just in case.

**Reply by JC5** [reply]
Ah. At the moment the UI doesn't email. This may be changed in the future.

**Reply by psychowood** [reply]
Oh sorry, I wasn't aware of that.

EDIT: Pushed a small PR to the docs.

**Reply by psychowood** [reply]
Since there is no report sent, would you consider adding something that allows to save the output messages in a local file? Something in plain html/javascript, no server side generation.

I can prepare a jquery script if it can be useful, it should have minimal impacts on the current page.

**Reply by JC5** [reply]
I added the report back to the UI routine, it should email now (in develop).

**Reply by psychowood** [reply]
Tested develop branch a few moments ago:
- email was sent (yay! 💯 )
- report has some issues, it looks like all three arrays (errors, warning, messages) are populated with the 9 messages I had discarding the 217 errors (lots of duplicated transactions)

Logs:
```
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Status: submission_running
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Messages: 9, warnings: 0, errors: 217
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Now in App\Services\Shared\Import\Status\SubmissionStatusManager::storeSubmissionStatus(submission-ikoUfurfyRWH): submission_done
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Messages: 9, warnings: 0, errors: 217
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Created event ImportedTransactions with filtering (2)
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Array contains 9 message(s)
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Array contains 9 warning(s)
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Array contains 9 error(s)
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Now in sendReportOverMail
fireflyiiiimporter    | [2024-01-04 18:46:16] local.INFO: Will send report message.
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Messages count: 9
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Warnings count: 9
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: Errors count  : 9
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: If no error below this line, mail was sent!
fireflyiiiimporter    | [2024-01-04 18:46:16] local.DEBUG: If no error above this line, mail was sent!
fireflyiiiimporter    | 172.23.0.28 - - [04/Jan/2024:18:45:33 +0100] "POST /import/submit/start?identifier=submission-ikoUfurfyRWH HTTP/1.1" 200 1150 "https://fireflyiiiimporter.lan/import/submit" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0"
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: isReadyForFileStep("submit")
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: isReadyForFileStep: Return true
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Now at App\Http\Controllers\Import\SubmitController::status(submission-ikoUfurfyRWH)
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Now in startOrFindJob(submission-ikoUfurfyRWH)
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Try to see if file exists for job submission-ikoUfurfyRWH.
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Status file exists for job submission-ikoUfurfyRWH.
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Status: submission_done
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Messages: 9, warnings: 0, errors: 217
fireflyiiiimporter    | 172.23.0.28 - - [04/Jan/2024:18:46:17 +0100] "GET /import/submit/status?identifier=submission-ikoUfurfyRWH HTTP/1.1" 200 27252 "https://fireflyiiiimporter.lan/import/submit" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0"
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: isReadyForFileStep("submit")
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: isReadyForFileStep: Return true
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Now at App\Http\Controllers\Import\SubmitController::status(submission-ikoUfurfyRWH)
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Now in startOrFindJob(submission-ikoUfurfyRWH)
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Try to see if file exists for job submission-ikoUfurfyRWH.
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Status file exists for job submission-ikoUfurfyRWH.
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Status: submission_done
fireflyiiiimporter    | [2024-01-04 18:46:17] local.DEBUG: Messages: 9, warnings: 0, errors: 217
fireflyiiiimporter    | 172.23.0.28 - - [04/Jan/2024:18:46:17 +0100] "GET /import/submit/status?identifier=submission-ikoUfurfyRWH HTTP/1.1" 200 27252 "https://fireflyiiiimporter.lan/import/submit" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0"
```

Partial (for privacy reasons) screen of the email 

![image](https://github.com/firefly-iii/firefly-iii/assets/115389/63cd11e5-f36a-4e70-9496-ac08241ae186)

**Reply by JC5** [reply]
Nice, I fixed that [three hours ago](https://github.com/firefly-iii/data-importer/commit/06b1a0e0f063f39305fcda270f32211611d474f0) 😁 , it should be part of the next Docker build tonight.

**Reply by psychowood** [reply]
We need instant CI builds :D (just joking ofc)
Thanks for the efforts.
